src/dbSta/src/dbNetwork.cc

```diff
@@ -121,36 +126,66 @@ class DbInstanceChildIterator : public InstanceChildIterator
 
  private:
   const dbNetwork* network_;
+  dbModule* module_;
   bool top_;
-  dbSet<dbInst>::iterator iter_;
-  dbSet<dbInst>::iterator end_;
+  dbSet<dbInst>::iterator dbinst_iter_;
+  dbSet<dbInst>::iterator dbinst_end_;
+  dbSet<dbModInst>::iterator modinst_iter_;
+  dbSet<dbModInst>::iterator modinst_end_;
 };
 
 DbInstanceChildIterator::DbInstanceChildIterator(const Instance* instance,
                                                  const dbNetwork* network)
     : network_(network)
 {
   dbBlock* block = network->block();
+  module_ = block->getTopModule();
+  modinst_iter_ = ((dbModule*) module_)->getModInsts().begin();
+  modinst_end_ = modinst_iter_;
```

Comment:
Why is end = begin?

Reply:
That is a little bit of paranoid programming to make sure that there is always a default for the end case even if the following branches fail so the code will work even if the instance is badly formed.